tests/utils/test_price_utils.py

```diff
@@ -946,3 +823,27 @@ def test_calc_k_const(self):
                 )
 
                 np.testing.assert_almost_equal(k, test_case["expected_result"], err_msg="unexpected k")
+
+
+class TestPriceUtils(BasePriceTest):
+    """Test calculations for each of the price utility functions"""
+
+    def test_calc_apr_from_spot_price(self):
+        """Execute the test"""
+        self.run_calc_apr_from_spot_price_test()
+
+    def test_calc_k_const(self):
+        """Execute the test"""
+        self.run_calc_k_const_test()
+
+    def test_calc_spot_price_from_apr(self):
+        """Execute the test"""
+        self.run_calc_spot_price_from_apr_test()
+
+    def test_calc_base_asset_reserves(self):
+        """Execute the test"""
+        self.run_calc_base_asset_reserves_test()
+
+    def test_calc_liquidity(self):
+        """Execute the test"""
+        self.run_calc_liquidity_test()
```

Comment:
I don't see the value in structuring our tests like this. Can you explain what the goal of this architecture is? Generally speaking, I see the value in parameterized test runners that can be composed to lead tests of emergent behavior, but I don't see that happening here.

Reply:
It's a boiler plate that has been useful elsewhere. For example, we can write a generic `calc_x_test` that receives a parameter, which is then looped in the `test_x` function. It's also nice for debugging because you can turn on/off tests easily. Elsewhere we use this structure to make it easier to turn on/off deletion of log files. You're correct that none of that is actually used here. I had a use set up in an earlier iteration of the refactor, but when we removed the element PM I no longer needed it.

Reply:
Ok, that all makes sense.

Reply:
e.g.:
``` 
    def test_calc_liquidity(self):
        """Execute the test"""
        self.run_calc_liquidity_test(sim_utils.get_pricing_model("hyperdrive"))
        self.run_calc_liquidity_test(sim_utils.get_pricing_model("yieldspace"))

```

Reply:
also here, in test_trade, where we run the same base test function with different arguments:
```
    def test_single_long(self):
        """Tests the BaseUser class"""
        self.run_base_trade_test(agent_policies=["single_long"], config_file="config/example_config.toml")

    def test_single_short(self):
        """Tests the BaseUser class"""
        self.run_base_trade_test(agent_policies=["single_short"], config_file="config/example_config.toml")
```